app/models/list_import/anime_planet/row.rb

```diff
@@ -0,0 +1,154 @@
+class ListImport
+  class AnimePlanet
+    class Row
+      attr_reader :node, :type
+
+      def initialize(node, type)
+        @node = node
+        @type = type
+      end
+
+      def media
+        key = "#{type}/#{media_info[:id]}"
+
+        Mapping.lookup('animeplanet', key) ||
+          Mapping.guess(type.classify.safe_constantize, media_info)
+      end
+
+      def media_info
+        {
+          id: node.attr('data-id')&.to_i,
+          title: extract_title(tooltip),
+          show_type: extract_show_type(tooltip),
+          episode_count: extract_total_episodes(tooltip),
+          chapter_count: extract_total_chapters(tooltip)
+        }.compact
+      end
+
+      def status
+        @status ||= extract_status(tooltip)
+
+        case @status
+        when /ing\z/ then :current
+        when /\AWant/ then :planned
+        when %r{Won't} then nil
+        when 'Stalled' then :on_hold
+        when 'Dropped' then :dropped
+        when 'Watched', /Read([0-9.]+)?/ then :completed
+        end
+      end
+
+      def progress
+        if status == (:completed || :planned || nil)
+          if type == 'anime'
+            amount = media_info[:episode_count]
```

Comment:
Yes, but if you have something on your `planned` list, it doesn't make sense to set your progress to 100%.  More likely, your progress is 0%